src/transform/src/equivalence_propagation.rs

```diff
@@ -490,10 +510,29 @@ impl EquivalencePropagation {
                 );
             }
             MirRelationExpr::TopK {
-                input, group_key, ..
+                input,
+                group_key,
+                limit,
+                ..
             } => {
-                // TODO: Update `limit` expressions, but only if we update `group_key` at the same time.
-                //       It is important to both or neither, to ensure that `limit` only references columns in `group_key`.
+                // We must be careful when updating `limit` to not install column references
+                // outside of `group_key`. We'll do this for now with `literal_domination`,
+                // which will ensure we only perform substitutions by a literal.
+                let input_equivalences = derived
+                    .last_child()
+                    .value::<Equivalences>()
+                    .expect("Equivalences required");
+                if let Some(input_equivalences) = input_equivalences {
+                    let input_types = derived
+                        .last_child()
+                        .value::<RelationType>()
+                        .expect("RelationType required");
+                    let reducer = input_equivalences.reducer();
+                    if let Some(expr) = limit {
+                        reducer.reduce_expr(expr);
+                        expr.reduce(input_types.as_ref().unwrap());
```

Comment:
Should there be a call to `literal_domination` here?

Reply:
Yes. I'll fix this up in post.